src/python.cpp

```diff
@@ -2207,3 +2207,91 @@ SEXP py_eval_impl(const std::string& code, bool convert = true) {
   return result;
 
 }
+
+// [[Rcpp::export]]
+SEXP py_convert_pandas_df(PyObjectRef obj) {
+  
+  // pd.DataFrame.items() iterates over (column name, Series) pairs
+  PyObjectPtr items(PyObject_CallMethod(obj, "items", NULL));
+
+  List list;
+  
+  while (true) {
+
+    // get next tuple
+    PyObjectPtr tuple(PyIter_Next(items));
+    if (tuple.is_null()) {
+      if (PyErr_Occurred())
+        stop(py_fetch_error());
+      else
+        break;
+    }
+    
+    // access Series in slot 1
+    PyObjectPtr series(PySequence_GetItem(tuple, 1));
+    
+    // and extract dtype
+    PyObjectPtr dtype(PyObject_GetAttrString(series, "dtype"));
+    PyObjectPtr dtype_str(PyObject_GetAttrString(dtype, "name"));
```

Comment:
nit: why not just call it `name` to match other object <-> attribute name mappings?

Reply:
check